packages/legend-studio/src/stores/editor-state/element-editor-state/RuntimeEditorState.ts

```diff
@@ -256,29 +277,28 @@ export const getRuntimeExplorerTreeData = (
       const childrenIds: string[] = [];
       if (store instanceof ModelStore) {
         // expand ModelStore to show source classes
-        const classes = uniq(
-          runtimeValue.allIdentifiedConnections
-            .filter(
-              (identifiedConnection) =>
-                identifiedConnection.connection.store.value instanceof
-                ModelStore,
-            )
-            .map((identifiedConnection) => {
-              const connectionValue =
-                identifiedConnection.connection instanceof ConnectionPointer
-                  ? identifiedConnection.connection.packageableConnection.value
-                      .connectionValue
-                  : identifiedConnection.connection;
-              if (
-                connectionValue instanceof JsonModelConnection ||
-                connectionValue instanceof XmlModelConnection
-              ) {
-                return connectionValue.class.value;
-              }
-              throw new UnsupportedOperationError();
-            })
-            .concat(allSourceClassesFromMappings),
-        ); // make sure we add classes (from mappings) that we expect to have connections for
+        let classes: Class[] = [];
+        runtimeValue.allIdentifiedConnections
+          .filter(
+            (identifiedConnection) =>
+              identifiedConnection.connection.store.value instanceof ModelStore,
+          )
+          .forEach((identifiedConnection) => {
+            const connectionValue =
+              identifiedConnection.connection instanceof ConnectionPointer
+                ? identifiedConnection.connection.packageableConnection.value
+                    .connectionValue
+                : identifiedConnection.connection;
+            if (
+              connectionValue instanceof JsonModelConnection ||
+              connectionValue instanceof XmlModelConnection
+            ) {
+              classes = classes.concat([connectionValue.class.value]);
+            }
+          });
```

Comment:
Code is kind of hard to read here. 
Could we go back to what it was but instead of `throw new UnsupportedOperationError()` returning undefined and then running a filter(isNonNullable) on the end.